transformer_engine/pytorch/attention.py

```diff
@@ -1656,6 +1657,9 @@ def __init__(
         assert (
             _flash_attn_version >= _flash_attn_version_required
         ), f"FlashAttention minimum version {_flash_attn_version_required} is required."
+        assert (
+            _flash_attn_version <= _flash_attn_max_version
+        ), f"FlashAttention maximum version {_flash_attn_max_version} is supported."
```

Comment:
Why do this?

Reply:
For consistency and adhering to the FA version bounds specified in setup.